src/ucs/sys/topo/base/topo.c

```diff
@@ -394,22 +429,27 @@ ucs_topo_get_distance_sysfs(ucs_sys_device_t device1,
     if (ucs_topo_is_pci_root(common_path)) {
         ucs_topo_set_distance(&ucs_global_opts.dist.phb,
                               ucs_topo_pci_root_bw(path1, path2), distance);
-        return UCS_OK;
+        goto out_free_buffers;
     } else if (ucs_topo_is_sys_root(common_path)) {
         if (ucs_topo_is_same_numa_node(device1, device2)) {
             ucs_topo_set_distance(&ucs_global_opts.dist.node, 17000 * UCS_MBYTE,
                                   distance);
-            return UCS_OK;
+            goto out_free_buffers;
         }
 
         ucs_topo_set_distance(&ucs_global_opts.dist.sys, 220 * UCS_MBYTE,
                               distance);
-        return UCS_OK;
+        goto out_free_buffers;
     }
 
     /* Report best perf for common PCI bridge or sysfs parsing error */
 default_distance:
-    return ucs_topo_get_distance_default(device1, device2, distance);
+    status = ucs_topo_get_distance_default(device1, device2, distance);
+out_free_buffers:
+    ucs_free(path1);
```

Comment:
They aren't initialized and can have arbitrary initial values. `free()` is called on them here..

Reply:
Initialized them to NULL.